compiler/src/iree/compiler/Dialect/Util/Conversion/ConversionPatterns.h

```diff
@@ -7,32 +7,72 @@
 #ifndef IREE_COMPILER_DIALECT_UTIL_CONVERSION_CONVERSIONPATTERNS_H_
 #define IREE_COMPILER_DIALECT_UTIL_CONVERSION_CONVERSIONPATTERNS_H_
 
+#include "llvm/ADT/SmallVector.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/BuiltinAttributes.h"
 #include "mlir/Pass/Pass.h"
+#include "mlir/Support/LLVM.h"
 #include "mlir/Transforms/DialectConversion.h"
 
 namespace mlir::iree_compiler {
 
-template <typename T>
+template <typename T, bool isTypeAttributeConversionEnabled = true>
```

Comment:
document what this parameter is for and why we would want it to be false

Reply:
I removed this argument. I was actually using it to test stuff. It seems that this does not introduce problems in other places. It also makes sense that the generic pattern would also handle types in attributes. If we require differentiation we can add it then.